agents/lakeshore240/LS240_agent.py

```diff
@@ -216,53 +254,91 @@ def stop_acq(self, session, params=None):
         else:
             return False, 'acq is not currently running'
 
-if __name__ == '__main__':
-    parser = site_config.add_arguments()
 
-    # Add options specific to this agent.
-    pgroup = parser.add_argument_group('Agent Options')
-    pgroup.add_argument('--serial-number')
-    pgroup.add_argument('--num-channels', default='2')
-    pgroup.add_argument('--mode')
-    pgroup.add_argument('--fake-data', default='0')
+def make_parser(parser=None):
+    if parser is None:
+        parser = argparse.ArgumentParser()
+
+    pgroup = parser.add_argument_group('Agent Config')
+    pgroup.add_argument('--serial-number', type=str,
+                        help="Serial number of your Lakeshore240 device")
+    pgroup.add_argument('--port', type=str,
+                        help="Path to USB node for the lakeshore")
+    pgroup.add_argument('--mode', type=str, choices=['idle', 'init', 'acq'],
+                        help="Starting action for the agent.")
+    pgroup.add_argument('--sampling-frequency', type=float,
+                        help="Sampling frequency for data acquisition")
+
+    return parser
+
+
+def main():
+    p = site_config.add_arguments()
+    parser = make_parser(parser=p)
+
+    #Not used anymore, but we don't it to break the agent if these args are passed
+    parser.add_argument('--fake-data')
+    parser.add_argument('--num-channels')
 
-    # Parse comand line.
     args = parser.parse_args()
 
+    if args.fake_data is not None:
+        print("WARNING: the --fake-data parameter is deprecated, please remove"
```

Comment:
There's a built in [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) that could be used instead of a print.